addons/payment/models/payment_token.py

```diff
@@ -113,10 +113,11 @@ def _get_available_tokens(
                 [('provider_id', 'in', providers_ids), ('partner_id', '=', partner_id)]
             )
         else:
+            partner = self.env['res.partner'].browse(partner_id)
             # Get all the tokens of the partner and of their commercial partner, regardless of
             # whether the providers are available.
-            partner = self.env['res.partner'].browse(partner_id)
-            return self.env['payment.token'].search(
+            # In sudo mode to access tokens of the commercial partner
```

Comment:
The call to this method is already sudoed in `PaymentPortal.payment_method`. I believe we should keep only the `sudo` in the controller, so that model methods remain "sudo agnostic". Pros are that it becomes clear that you deal with sudoed recordset in the controller, and the model doesn't need to consider access rights; cons are that the `sudo` is done a bit far from the domain that requires the sudo.